components/script/dom/bindings/settings_stack.rs

```diff
@@ -64,7 +77,81 @@ impl Drop for AutoEntryScript {
 pub fn entry_global() -> Root<GlobalScope> {
     STACK.with(|stack| {
         stack.borrow()
-             .last()
+             .iter()
+             .rev()
+             .find(|entry| entry.kind == StackEntryKind::Entry)
              .map(|entry| Root::from_ref(&*entry.global))
     }).unwrap()
 }
+
+/// RAII struct that pushes and pops entries from the script settings stack.
+pub struct AutoIncumbentScript {
```

Comment:
There might be a better way to write it; the important thing is that we never dereference the field, we just compare / print the address.